### PR TITLE
allow customizing favicon

### LINKED ISSUE #72: Allow customizing favicon
~~Add `faviconPath` to `neuron.dhall` configuration, with the default using neuron's favicon.~~ See https://github.com/srid/neuron/issues/72#issuecomment-650777042

### PR DESCRIPTION
**Pull request by smunix**
fix #72

### CODE REVIEW

File: neuron/src-dhall/Config/Type.dhall [hunk 1]
```diff
@@ -14,4 +14,6 @@
     Bool
 , minVersion :
     Text
+, faviconPath :
```
Comment (srid) [inline neuron/src-dhall/Config/Type.dhall:17]: Rename from `faviconPath` to `favicon`

File: guide/2011701.md [hunk 2]
```diff
@@ -4,15 +4,15 @@ You may configure the parameters of your web interface by adding an optional con
...
+| **`siteBaseUrl`** | The base URL of your published Neuron site. Setting the base URL will enable [breadcrumbs](https://developers.google.com/search/docs/data-types/breadcrumb) in your site's structured data |
+| **`theme`**       | Color scheme to use for your site. Value must be [one of the color names](https://semantic-ui.com/usage/theming.html#sitewide-defaults) supported by SemanticUI.                           |
+| **`aliases`**     | Setup custom redirects in the statically generated site.                                                                                                                                   |
+| **`editUrl`**     | The URL (without the zettel filename) to edit zettels.                                                                                                                                     |
+| **`minVersion`**  | The minimum neuron version your site must be generated with.                                                                                                                               |
```
[the remark this answers is not in the document]
Comment (smunix) [reply guide/2011701.md:15]: I used emacs, markdown-mode.

File: neuron/src/app/Neuron/Web/View.hs [hunk 3]
```diff
@@ -67,7 +67,7 @@ renderRouteHead config route val = do
   elAttr "meta" ("http-equiv" =: "Content-Type" <> "content" =: "text/html; charset=utf-8") blank
   elAttr "meta" ("name" =: "viewport" <> "content" =: "width=device-width, initial-scale=1") blank
   el "title" $ text $ routeTitle config (snd val) route
-  elAttr "link" ("rel" =: "shortcut icon" <> "href" =: "https://raw.githubusercontent.com/srid/neuron/master/assets/neuron.svg") blank
+  elAttr "link" ("rel" =: "shortcut icon" <> "href" =: (maybe "https://raw.githubusercontent.com/srid/neuron/master/assets/neuron.svg" id $ faviconPath config)) blank
```
[the remark this answers is not in the document]
Comment (smunix) [reply neuron/src/app/Neuron/Web/View.hs:70]: true! let me fix it

File: neuron/src/app/Neuron/Web/View.hs [hunk 4]
```diff
@@ -67,7 +67,7 @@ renderRouteHead config route val = do
   elAttr "meta" ("http-equiv" =: "Content-Type" <> "content" =: "text/html; charset=utf-8") blank
   elAttr "meta" ("name" =: "viewport" <> "content" =: "width=device-width, initial-scale=1") blank
   el "title" $ text $ routeTitle config (snd val) route
-  elAttr "link" ("rel" =: "shortcut icon" <> "href" =: "https://raw.githubusercontent.com/srid/neuron/master/assets/neuron.svg") blank
+  elAttr "link" ("rel" =: "shortcut icon" <> (maybe mempty ("href" =:) $ faviconUrl config)) blank
```
Comment (srid) [inline neuron/src/app/Neuron/Web/View.hs:70]: This is better:

```haskell
forM_ (faviconUrl config) $ \href ->
  elAttr "link" ("rel" =: "shortcut icon" <> "href" =: href) blank
```

File: neuron/src/app/Neuron/Web/View.hs [hunk 5]
```diff
@@ -67,7 +67,9 @@ renderRouteHead config route val = do
   elAttr "meta" ("http-equiv" =: "Content-Type" <> "content" =: "text/html; charset=utf-8") blank
   elAttr "meta" ("name" =: "viewport" <> "content" =: "width=device-width, initial-scale=1") blank
   el "title" $ text $ routeTitle config (snd val) route
-  elAttr "link" ("rel" =: "shortcut icon" <> "href" =: "https://raw.githubusercontent.com/srid/neuron/master/assets/neuron.svg") blank
+  case faviconUrl config of
+    Just iconUrl -> elAttr "link" ("rel" =: "shortcut icon" <> "href" =: iconUrl) blank
+    _ -> blank
```
Comment (srid) [inline neuron/src/app/Neuron/Web/View.hs:72]: Oh, there is `whenJust` from relude:

```haskell
whenJust (faviconUrl config) $ \href -> 
  elAttr "link" ("rel" =: "shortcut icon" <> "href" =: href) blank
```

**Comment by srid** [comment]
@smunix I am closing this, because we should approach the problem from a more general angle. See https://github.com/srid/neuron/issues/72#issuecomment-650777042

**Pull request closed without merging**